d2l-input-shared-styles.js

```diff
@@ -105,8 +105,8 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-styles">
 			textarea.d2l-input:hover:disabled {
 				@apply --d2l-input-hover-disabled;
 			}
-			input.d2l-input:hover, input.d2l-input:focus,
-			textarea.d2l-input:hover, textarea.d2l-input:focus {
+			input.d2l-input:hover:not(:disabled), input.d2l-input:focus:not(:disabled),
```

Comment:
Could these `:not` selectors be avoided by adding a hover/focus selector further below?
```
input.d2l-input:disabled,
input.d2l-input:hover:disabled,
input.d2l-input:focus:disabled,
textarea.d2l-input:disabled,
textarea.d2l-input:hover:disabled,
textarea.d2l-input:focus:disabled {
     @apply --d2l-input-disabled;
}
```

Reply:
@mdgbayly : I think some degree of duplication will likely be needed here.  I was thinking it would be good to avoid the `:not` if possible.  Using `!important` may also be possible, but I usually only go that route as an absolute last resort.

Reply:
@dbatiste Do you recall why we have the type attribute selectors?
e.g. `input[type="text"].d2l-input`.
It seems to be this which is throwing off the ShadyCSS scoping shim.

Reply:
Oh, I guess because of type="button" and maybe others...

Reply:
I will update the PR with duplicate type attribute selectors.